register/forms.py

```diff
@@ -33,6 +33,12 @@ def __init__(self, *args, **kwargs):
             {"placeholder": "Password (Once more)"}
         )
 
+    def clean_username(self):
+        username = self.cleaned_data["username"]
+        if Student.objects.filter(username=username).exists():

```

Comment:
you can use get() as it should return only one object as username is unique.

Reply:
when I switched it to get() a lot of the tests failed. any idea why?

Reply:
Get is meant to return single objects mostly used when to filter on primary key, where as filter() returns multiple objects. You will need to modify the test cases accordingly